src/main/java/com/es/phoneshop/utils/ParseValidator.java

```diff
@@ -0,0 +1,12 @@
+package com.es.phoneshop.utils;
+
+public class ParseValidator {
```

Comment:
generally it should be a singleton and implement some interface
errorMap should be a parameter of such Validator and it can be updated here in case of invalid input